ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/GetBulk.java

```diff
@@ -116,88 +134,96 @@ public CliCommand init(final Arguments args) throws Exception {
 
     @Override
     public DefaultResult call() throws Exception {
-        final Ds3ClientHelpers.ObjectChannelBuilder getter
-                = this.discard ?  new MemoryObjectChannelBuilder(DEFAULT_BUFFER_SIZE, DEFAULT_FILE_SIZE)
-                : new FileObjectGetter(this.outputPath);
+        final Ds3ClientHelpers.ObjectChannelBuilder getter;
+        if (this.pipe) {
+            getter = new PipedFileObjectGetter(this.outputPath, FileUtils.normalizedObjectNames(this.pipedFileNames));
+        } else if (this.discard) {
+            getter = new MemoryObjectChannelBuilder();
+        } else {
+            getter = new FileObjectGetter(this.outputPath);
+        }
+        LOG.info(getLogDescription());
+        return new DefaultResult(this.restore(getter));
+    }
 
+    // preserve legacy descriptions from different code paths
+    private String getLogDescription() {
+        if (this.pipe) {
+            return "Getting piped objects from " + this.bucketName;
+        }
         if (this.sync) {
             if (Guard.isNullOrEmpty(this.prefixes)) {
-                LOG.info("Syncing all objects from {}", this.bucketName);
+                return "Syncing all objects from " + this.bucketName;
             } else {
-                LOG.info("Syncing only those objects that start with {}", Joiner.on(" ").join(this.prefixes));
+                return "Syncing only those objects that start with " + Joiner.on(" ").join(this.prefixes);
             }
-            return new DefaultResult(this.restoreSome(getter));
         }
-
-        if (!Guard.isNotNullAndNotEmpty(prefixes)) {
-            LOG.info("Getting all objects from {}", this.bucketName);
-            return new DefaultResult(this.restoreAll(getter));
+        if (Guard.isNotNullAndNotEmpty(prefixes)) {
+            return "Getting only those objects that start with " + Joiner.on(" ").join(this.prefixes);
         }
-
-        LOG.info("Getting only those objects that start with {}", Joiner.on(" ").join(this.prefixes));
-        return new DefaultResult(this.restoreSome(getter));
+        return "Getting all objects from " + this.bucketName;
     }
 
-    private String restoreSome(final Ds3ClientHelpers.ObjectChannelBuilder getter) throws IOException, XmlProcessingException {
+    private String restore(final Ds3ClientHelpers.ObjectChannelBuilder getter) throws CommandException, IOException, XmlProcessingException {
```

Comment:
I would like to see less special casing in this one method.  The special casing makes it difficult to follow.  My suggestion would be to create different restore methods, one for piped, synced, or whatever else we need.